### PR TITLE
feat: add more test

### PR DESCRIPTION
**Pull request by MussinaLin**
Add more tests for testing

- agent initialize
- router setting fee collector
- router setting fee calculator

### CODE REVIEW

File: test/fees/FeeGeneralCases.t.sol [hunk 1]
```diff
@@ -115,6 +118,59 @@ contract FeeRevertCasesTest is Test {
...
+    }
+
+    function testSetFeeCalculators() external {
+        bytes4[] memory selectors = new bytes4[](2);
+        selectors[0] = AAVE_FLASHLOAN_SELECTOR;
+        selectors[1] = NATIVE_FEE_SELECTOR;
+
+        address[] memory tos = new address[](2);
+        tos[0] = v2Pool;
+        tos[1] = DUMMY_TO_ADDRESS;
```
Comment (Jeff-CCH) [inline test/fees/FeeGeneralCases.t.sol:134]: Please rebase master to fix the naming

Comment (zodahu) [reply test/fees/FeeGeneralCases.t.sol:134]: FYI, I renamed to `ANY_TO_ADDRESS`

File: test/Agent.t.sol [hunk 2]
```diff
@@ -356,4 +356,9 @@ contract AgentTest is Test {
         vm.prank(router);
         agent.execute(logics, feesEmpty, tokensReturnEmpty);
     }
+
+    function testCannotInitializeTwice() external {
+        vm.expectRevert(IAgent.Initialized.selector);
+        agent.initialize();
+    }
```
[the remark this answers is not in the document]
Comment (zodahu) [reply test/Agent.t.sol:363]: Since the PR has been merged, please help remove this test. Thanks.

File: test/fees/FeeGeneralCases.t.sol (hunk 1 above)
Comment (MussinaLin) [reply test/fees/FeeGeneralCases.t.sol]: fix, thanks

File: test/Agent.t.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (MussinaLin) [reply test/Agent.t.sol]: Deleted. thanks

File: test/fees/FeeGeneralCases.t.sol [hunk 3]
```diff
@@ -115,6 +118,59 @@ contract FeeRevertCasesTest is Test {
...
+
+        feeCalculators = new address[](1);
+        feeCalculators[0] = address(flashLoanFeeCalculator);
+
+        vm.expectRevert(FeeGenerator.LengthMismatch.selector);
+        router.setFeeCalculators(selectors, tos, feeCalculators);
+    }
+
```
[the remark this answers is not in the document]
Comment (MussinaLin) [reply test/fees/FeeGeneralCases.t.sol]: Simplified. thanks

File: test/fees/FeeGeneralCases.t.sol [hunk 4]
```diff
@@ -115,6 +118,46 @@ contract FeeRevertCasesTest is Test {
         router.execute{value: nativeAmount}(logics, tokensReturnEmpty, SIGNER_REFERRAL);
     }
 
+    function testSetFeeCalculator(bytes4 selector, address to, address feeCalculator) external {
```
Comment (zodahu) [inline test/fees/FeeGeneralCases.t.sol:121]: Should test `onlyOwner` so need to add `testSetFeeCalculatorByNonOwner` and `testSetFeeCalculatorsByNonOwner`

Comment (MussinaLin) [reply test/fees/FeeGeneralCases.t.sol:121]: Added. Saw other `xxxByNonOwner` tests  use `Cannot` prefix, so add `Cannot` as well.

**Comment by zodahu** [comment]
LGTM

**Comment by Jeff-CCH** [comment]
LGTM

**Pull request merged**